nft-client/src/main.rs

```diff
@@ -0,0 +1,635 @@
+    }
+}
+
+/// Display a contract address using the <index,subindex> notation where index
+/// and subindex are replaced with an u64.
+impl Display for ContractAddressWrapper {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+        write!(f, "<{},{}>", self.0.index.index, self.0.subindex.sub_index)
+    }
+}
+
+/// Name of a contract receive function, wrapped to implement FromStr.
+#[derive(Debug, Clone)]
+pub struct OwnedReceiveNameWrapper(concordium_contracts_common::OwnedReceiveName);
+
+/// Parses a contract receive function name.
+impl FromStr for OwnedReceiveNameWrapper {
+    type Err = concordium_contracts_common::NewReceiveNameError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(OwnedReceiveNameWrapper(
+            concordium_contracts_common::OwnedReceiveName::new(s.to_string())?,
+        ))
+    }
+}
+
+/// Wrapper for Address to implement FromStr and Display using for strings which
+/// are either an account address or contract address.
+#[derive(Debug)]
+struct AddressWrapper(types::Address);
+
+/// Parse a string into an address, by first trying to parse the string as a
+/// contract address string, otherwise try parsing as an account address string.
+impl FromStr for AddressWrapper {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let contract_result = ContractAddressWrapper::from_str(s);
+        let address = if let Ok(contract) = contract_result {
+            types::Address::Contract(contract.0)
+        } else {
+            types::Address::Account(id::types::AccountAddress::from_str(s)?)
+        };
+        Ok(AddressWrapper(address))
+    }
+}
+
+/// Display the Address using contract notation <index,subindex> for contract
+/// addresses and display for account addresses.
+impl Display for AddressWrapper {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+        match self.0 {
+            types::Address::Account(a) => a.fmt(f),
+            types::Address::Contract(c) => ContractAddressWrapper(c).fmt(f),
+        }
+    }
+}
+
+/// Helper to parse account keys JSON format.
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct AccountData {
+    /// The keys for the account address.
+    account_keys: id::types::AccountKeys,
+    /// The account address.
+    address:      id::types::AccountAddress,
+}
+
+/// Structure to hold command-line arguments.
+#[derive(StructOpt)]
+#[structopt(
+    bin_name = "nft-client",
+    about = "NFT client tool for interacting with CTS1 NFT token contracts on the Concordium \
+             blockchain"
+)]
+struct App {
+    #[structopt(
+        long = "node",
+        help = "GRPC interface of the node.",
+        default_value = "http://localhost:10001"
+    )]
+    endpoint: Endpoint,
+    #[structopt(subcommand)]
+    command:  Command,
+}
+
+#[derive(Debug, StructOpt)]
+#[allow(clippy::large_enum_variant)]
```

Comment:
This is almost always a bad idea, and should be refactored by boxing the offending variants.